ecl/hql/hqlgram.y

```diff
@@ -8812,6 +8812,10 @@ simpleDataSet
 
                             bool hasFileposition = getBoolAttributeInList(extra, filepositionAtom, true);
                             record.setown(parser->checkIndexRecord(record, $5, extra));
+
+                            if (!parser->checkAllowed($1, "datafile", "Datafile access"))
+                                extra.setown(createComma(extra.getClear(), createAttribute(_disallowed_Atom)));
```

Comment:
There's probably a good reason for it and I see PIPE is checked at parse time too..
Is it unavoidable to flag disallowed here, rather than when the error will actually be hit later when/if the attribute is used, e.g. in doBuildActivityDiskRead.

@shamser @ghalliday